### PR TITLE
Add Dockerfile for concourse-ci linux-32

### PR DESCRIPTION
**Pull request by nehaljwani**
CC @msarahan @mingwandroid @kalefranz

### CODE REVIEW

File: docker-32/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+#RUN mkdir -p bin && \
+#  LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' https://github.com/concourse/concourse/releases/latest) && \
+#  LATEST_VERSION=$(echo $LATEST_RELEASE | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/') && \
+#  ARTIFACT_URL="https://github.com/concourse/concourse/releases/download/$LATEST_VERSION/fly_linux_amd64" && \
+#  curl -L ${ARTIFACT_URL} -o bin/fly && chmod +x bin/fly
+
+# This fly binary has been compiled manually
```
Comment (msarahan) [inline docker-32/Dockerfile]: Could we please document more on how this was done?  Ideally, I'd do the build in this dockerfile, but I understand that might be difficult or overly verbose.

**Comment by nehaljwani** [comment]
@msarahan I have added the script to build the fly 32-bit binary and now it will be built each time the container is built. Now the only thing we need to fix is:
```
PackageNotFoundError: Package missing in current linux-32 channels: 
  - conda-concourse-ci
```

**Comment by msarahan** [comment]
Excellent, nice work. I will use this container to build the initial version of c3i for Linux 32 today.

File: docker-32/install_miniconda.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+rm $fname
+/opt/miniconda/bin/conda config --set show_channel_urls True
+# update --all is pulling in replacements from defaults
+# /opt/miniconda/bin/conda update --yes --all
+/opt/miniconda/bin/conda config --add channels c3i_test
+/opt/miniconda/bin/conda install --yes git conda-build curl anaconda-client
+/opt/miniconda/bin/conda install --yes conda-concourse-ci
+/opt/miniconda/bin/conda config --set add_pip_as_python_dependency False
+/opt/miniconda/bin/conda config --set anaconda_upload True
+/opt/miniconda/bin/conda clean -ptiy
```
Comment (kalefranz) [inline docker-32/install_miniconda.sh]: Let's make the canonical location for the root env everywhere on all unix from here on out `/opt/conda`.  That's what I'm trying to standardize on.

**Comment by kalefranz** [comment]
Let's make two directories within the root `docker` directory in this repo.  The current contents in master within the `docker` directory should go into a directory named `c3i-linux-64`, since this is the base image for the `linux-64` subdir, and the new `Dockerfile` and other work here should go into `c3i-linux-32`.  The image names can then match those directory names.

File: docker-32/install_miniconda.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+fname=$(ls Miniconda3-*-x86.sh | tail -n 1)
+# this is Ray Donnelly's custom Miniconda, built with our new gcc toolchain
+bash -x $fname -bfp /opt/miniconda
+rm $fname
+/opt/miniconda/bin/conda config --set show_channel_urls True
+# update --all is pulling in replacements from defaults
+# /opt/miniconda/bin/conda update --yes --all
+/opt/miniconda/bin/conda config --add channels c3i_test
+/opt/miniconda/bin/conda install --yes git conda-build curl anaconda-client
+/opt/miniconda/bin/conda install --yes conda-concourse-ci
```
Comment (kalefranz) [inline docker-32/install_miniconda.sh]: For now, until we get the 32-bit package built, this line can probably be

    # conda-install c3i dependencies, then pip-install c3i
    /opt/conda/bin/conda install -y contextlib2 networkx setuptools six yaml
    /opt/conda/bin/pip install https://github.com/conda/conda-concourse-ci/archive/master.zip

File: docker-32/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  patch \
+  rsync \
+  util-linux \
+  which \
+  wget \
+  && yum clean all
+
+WORKDIR /build_scripts
+COPY install_miniconda.sh /build_scripts
+COPY Miniconda3-* /build_scripts/
```
Comment (kalefranz) [inline docker-32/Dockerfile]: We obviously don't want the Miniconda.sh file checked into this repo.  Let's curl it into the image from https://repo.continuum.io/pkgs/misc/preview/

File: docker-32/build_concourse_fly.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+OLDDIR="${PWD}"
+CONCOURSE_FLY_VERSION="3.3.0"
```
Comment (kalefranz) [inline docker-32/build_concourse_fly.sh]: Good.  We'll probably update this soon.  But this is perfect right now.

File: docker-32/install_miniconda.sh (hunk 3 above)
Comment (nehaljwani) [inline docker-32/install_miniconda.sh]: Done.

File: docker-32/Dockerfile (hunk 4 above)
Comment (nehaljwani) [inline docker-32/Dockerfile]: Done.

File: docker-32/build_concourse_fly.sh (hunk 5 above)
Comment (nehaljwani) [inline docker-32/build_concourse_fly.sh]: Yay!

**Comment by nehaljwani** [comment]
Merging for now, will make PRs for changes as needed.

**Pull request merged**